bitcoin/script/miniscript.h

```diff
@@ -765,7 +765,10 @@ struct Node {
     bool NeedsSignature() const { return GetType() << "s"_mst; }
 
     //! Do all sanity checks.
-    bool IsSafeTopLevel() const { return GetType() << "Bms"_mst && CheckOpsLimit() && CheckStackSize(); }
+    bool IsSane() const { return GetType() << "m"_mst && CheckOpsLimit() && CheckStackSize() && IsValid(); }
```

Comment:
Now that there is an `IsSane()`, perhaps all the ad-hoc sanity checks in the compiler can be replaced with just a call to this? (see `Compilation::Add`).